platform/kaniko_cache.go

```diff
@@ -0,0 +1,107 @@
+package platform
+
+import (
+	"context"
+	"fmt"
+	"github.com/kiegroup/container-builder/client"
+	"github.com/kiegroup/container-builder/util/defaults"
+	v08 "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+	"github.com/kiegroup/kogito-serverless-operator/builder"
+	"github.com/kiegroup/kogito-serverless-operator/constants"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func createKanikoCacheWarmerPod(ctx context.Context, client client.Client, platform *v08.KogitoServerlessPlatform) error {
+	// The pod will be scheduled to nodes that are selected by the persistent volume
+	// node affinity spec, if any, as provisioned by the persistent volume claim storage
+	// class provisioner.
+	// See:
+	// - https://kubernetes.io/docs/concepts/storage/persistent-volumes/#node-affinity
+	// - https://kubernetes.io/docs/concepts/storage/volumes/#local
+	// nolint: staticcheck
+	pvcName := constants.DEFAULT_KANIKOCACHE_PVC_NAME
+	if persistentVolumeClaim, found := platform.Status.BuildPlatform.PublishStrategyOptions[builder.KanikoPVCName]; found {
+		pvcName = persistentVolumeClaim
+	}
+
+	var warmerImage string
+	if image, found := platform.Status.BuildPlatform.PublishStrategyOptions[builder.KanikoWarmerImage]; found {
+		warmerImage = image
+	} else {
+		warmerImage = fmt.Sprintf("%s:v%s", builder.KanikoDefaultWarmerImageName, defaults.KanikoVersion)
+	}
+
+	pod := corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: platform.Namespace,
+			Name:      platform.Name + "-cache",
+			Labels: map[string]string{
+				"camel.apache.org/component": "kaniko-warmer",
```

Comment:
same here, maybe add info about the workflow that is being built?

Reply:
The cache will be share between all the build on the same platform (i.e. namespace) so it isn't related to a specific workflow